mmcv/transforms/processing.py

```diff
@@ -1317,6 +1319,7 @@ def _random_sample_ratio(
         scale = int(scale[0] * ratio), int(scale[1] * ratio)
         return scale
 
+    @cacheable_method
```

Comment:
Maybe to decorate `_random_sample_ratio` and `_random_sample` to save the `scale`